### DISCUSSION TITLE
push fails, asks for userid/password when ssh is setup and works elsewhere

### DISCUSSION THREAD

**Discussion by pfarrell51** [community/community#38388] [Other Feature Feedback, Questions, & Ideas] [answered]
I have not tried to push to my personal github account in 3 or 4 years.
I am running on a windows 11 computer using WSL (windows system for linux) shell

I'm getting a failure trying to do the usual "git push origin master"
I have had my ISP's router replaced, and perhaps it is not passing SSL out, but
I have used SSL successfully for other applications (such as rsync'ing to my webhost's servers).

When I do a simple git push, it asks for user id and password
pfarrell@Alien15:~/pfarrell51/cmd$ git push origin master
Username for 'https://github.com':
Password for 'https://github.com':
remote: No anonymous write access.
fatal: Authentication failed for 'https://github.com/pfarrell51/gows/'

When I enter the proper userid and password, it fails saying support was removed over a year ago.
pfarrell@Alien15:/pfarrell51/cmd$ git push origin master
Username for 'https://github.com': pfarrell51
Password for 'https://pfarrell51@github.com':
remote: Support for password authentication was removed on August 13, 2021.
remote: Please see https://docs.github.com/en/get-started/getting-started-with-git/about-remote-repositories#cloning-with-https-urls for information on currently recommended modes of authentication.
fatal: Authentication failed for 'https://github.com/pfarrell51/gows/'

I can use the ssh comman to access github it works and says my ssh keys are allowing me to be
authenticaled:
/pfarrell51/cmd$
pfarrell@Alien15:~/whome/sandbox/gows/src/github.com/pfarrell51/cmd$ ssh -T git@github.com
Hi pfarrell51! You've successfully authenticated, but GitHub does not provide shell access.

When I follow the network/ssh debugging examples, it fails network is unreachable

pfarrell@Alien15:~/whome/sandbox/gows/src/github.com/pfarrell51/cmd$ ssh -T 443 git@ssh.github.com
ssh: connect to host 443 port 22: Network is unreachable

when I use the verbose option, it still fails
/pfarrell51/cmd$ git push origin master
Username for 'https://github.com': pfarrell51
Password for 'https://pfarrell51@github.com':
remote: Support for password authentication was removed on August 13, 2021.
remote: Please see https://docs.github.com/en/get-started/getting-started-with-git/about-remote-repositories#cloning-with-https-urls for information on currently recommended modes of authentication.
fatal: Authentication failed for 'https://github.com/pfarrell51/gows/'

I've even installed the github desktop GUI application, but I can't get it to work either. It is installed but can't seem to connect to my account's existing repositories.

I'm at a loss for how to proceed to do problem isolation on this.
Thanks
Pat

**Comment by airtower-luna** [comment, marked as answer]
This shows Git is trying to use HTTPS, not SSH:

> `fatal: Authentication failed for 'https://github.com/pfarrell51/gows/'`

Which protocol gets used depends on the configured URL. You can set the correct (SSH) URL with [`git remote set-url`](https://git-scm.com/docs/git-remote#Documentation/git-remote.txt-emset-urlem):

```
git remote set-url origin git@github.com:pfarrell51/gows.git
```

After that Git should use SSH.

**Comment by pfarrell51** [comment]
Thank you so much, that worked perfectly.

**Comment by Uudukku** [comment]
![IMG_20221201_233914](https://user-images.githubusercontent.com/111229013/205431452-db9d196d-0113-471b-967f-3e97338b6fe8.jpg)
My Sandbox Failed.

**Comment by Rikesh307** [comment]
I forgot my user and password how can i push my code now can i get solution please?

**Reply by airtower-luna** [reply]
See here: https://docs.github.com/en/site-policy/other-site-policies/github-account-recovery-policy

Mind that you cannot push using your password either way, for that you need an SSH key or for HTTPS a token with appropriate permissions.
